src/shared/redux/slices/publicationsSlice.ts

```diff
@@ -1,31 +1,58 @@
-import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { createSlice, current, PayloadAction } from '@reduxjs/toolkit';
 import { RootState } from '../rootReducer';
 
-type CollaboratorElement = {
-  github_username: string;
+type Collaborator = {
+  id: string;
+  githubUsername: string;
   role: string;
 };
 
-type PublicationListElement = {
-  project_name: string;
-  collaborators: [CollaboratorElement];
-  pm_preference: string;
-  description: string;
+type Publication = {
+  id: string;
   dirPath: string;
+  publicationName: string;
+  description: string;
+  collaborators: Collaborator[];
+  packageManager: string;
+  useSass: string;
+  useTypescript: string;
 };
 
 type PublicationList = {
-  list: [PublicationListElement];
+  list: Publication[];
+};
+
+// type created to exclude collaborator type to avoid an issue with
+// index access writes on union of keys, which would complicate the
+// code significantly more
+type PublicationPrimitives = Omit<Publication, 'collaborators'>;
+
+// type created to define payload structure for actions on
+// publication primitive fields
+type PublicationModification = {
+  id: string;
+  field: keyof PublicationPrimitives;
+  value: string;
+};
+
+// type created to define payload structure for actions on
+// publication collaborators
+type CollaboratorListModification = {
+  pubId: string;
+  value: Collaborator | string;
```

Comment:
The typing for this action might be a little confusing. It will allow the developer to pass in a value of string for when he wants to add a collaborator (not only when deleting) and therefore just push a string to collaborators array. Perhaps make it into a generic.

```ts
interface CollaboratorListModification<T> {
  id: string;
  value: T
}
```

Then you can pass it in to the action definitions as `CollaboratorListModification<Collaborator>` and `CollaboratorListModification<string>`